storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -142,6 +147,15 @@ public void run() {
                 
             }
         };
+
+        connector = new Runnable() {
+            @Override
+            public void run() {
+                if (!closing) {
+                    connect();
+                }
+            }
+        };
         
         long initialDelay = Math.min(30L * 1000, max_sleep_ms * max_retries); //max wait for 30s
```

Comment:
we should use Math.max(...)?
No, I do not think so. because .max returns the argument with the greatest value.
So Math.min (30L * 1000 * max_sleep_ms max_retries) will return 30 seconds if "max_sleep_ms * max_retries" is greater than 30 s! and vice versa